user_sync/app.py

```diff
@@ -353,6 +359,17 @@ def begin_work(config_loader):
     directory_groups = config_loader.get_directory_groups()
     rule_config = config_loader.get_rule_options()
 
+    ssl_cert_verify = rule_config.get('ssl_cert_verify', True)
+    user_sync.connector.umapi.UmapiConnector.ssl_cert_verify = ssl_cert_verify
```

Comment:
If we modifythe `ssl_cert_verify` class property here, it will change that value for all `UmapiConnector` objects, and we may not always want to do that in the future. I recommend passing `ssl_cert_verify` as a new parameter to the `UmapiConnector` constructor and setting it as an object attribute.

Reply:
Sure - I can make this change.  I intentionally chose class properties because setting this value should be consistent across all the affected modules, regardless of how they  are instantiated.  It doesn't make sense to apply it to a single object but not another one.  Sync would not succeed unless they were all disabled (the driving force for this PR).  Using class properties was the simplest way to accomplish this.  I will rework it as an arg instead for primary/secondaries, sign sync, and console connector.

Reply:
Thanks, It isn't as much about how we do things now vs. how things may change in the future. In general, changing things that can affect the scope beyond a single object or variable can create bugs that are difficult to track down.